### PR TITLE
Implement fimdb helper like a wrapper between new design and old design

### LINKED ISSUE #10769: Implement FIMDBHelper namespace
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/10227 | FIM |

### Description

This issue intends to create `FimDBHelper` namespace to integrate old design with the new design of FIM DB. It implies:

- Every function used externally should be included in the `FimDBHelper` which acts like a translator between DBSync, files and registries.
- Every query should be replaced with equivalent operations performed by `dbsync`
- Every added method should be accompanied by unit tests.

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

This pull request adds the following changes:

- [x] Implement FIMDBHelper namespace
- [x] Add  unit tests to validate this development

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Unit tests passed
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] cppcheck (static code analysis)

Closes #10769

### CODE REVIEW

File: src/syscheckd/db/include/fimDBHelper.hpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    /**
+    * @brief Update a row from a table.
+    *
+    * @param tableName a string with the table name
+    * @param item a RegistryKey, RegistryValue or File with their parameters
+    *
+    * @return 0 on success, another value otherwise.
+    */
+    int updateItem(const std::string &, const nlohmann::json &);
```
[the remark this answers is not in the document]
Comment (FrancoRivero) [reply src/syscheckd/db/include/fimDBHelper.hpp]: Hi @JoseAntonioMHerrera, I am adding this change in this file !

File: src/syscheckd/db/include/commonDefs.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#include "logging_helper.h"
+
+enum dbQueryResult
+{
+    SUCCESS,
+    MAX_ROWS_ERROR,
+    DBSYNC_ERROR
+};
+
+typedef void((*fim_sync_callback_t)(const char *tag, const char* buffer));
```
Comment (FrancoRivero) [inline src/syscheckd/db/include/commonDefs.h:20]: Could you add a name to this typedef to simplify the function call?

File: src/syscheckd/db/include/commonDefs.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+typedef void((*logging_callback_t)(const modules_log_level_t level, const char* log));
```
Comment (FrancoRivero) [inline src/syscheckd/db/include/commonDefs.h:21]: Could you add a name to this typedef to simplify the function call?

**Review by chemamartinez — APPROVED** [review]

**Review by JoseAntonioMHerrera — APPROVED** [review]
LGTM!

**Pull request merged**